src/ValueObject/ImageValueObject.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme\ValueObject;
+
+use Drupal\image\Plugin\Field\FieldType\ImageItem;
+
+/**
+ * Handle information about an image media item.
+ */
+class ImageValueObject extends ValueObjectBase {
+
+  /**
+   * Media Source.
+   *
+   * @var string
+   */
+  protected $src;
+
+  /**
+   * The alt of the image.
+   *
+   * @var string
+   */
+  protected $alt;
+
+  /**
+   * The name of the media.
+   *
+   * @var string
+   */
+  protected $name;
+
+
+  /**
+   * The parameter 'responsive' of the image.
+   *
+   * @var bool
+   */
+  protected $responsive;
+
+  /**
+   * ImageValueObject constructor.
+   *
+   * @param string $src
+   *   Media URL, including Drupal schema if internal.
+   * @param string $alt
+   *   Image alt text.
+   * @param string $name
+   *   Name of the image, e.g. "example.jpg".
+   * @param bool $responsive
+   *   Responsiveness of the image.
+   */
+  private function __construct(string $src, string $alt = '', string $name = '', bool $responsive = TRUE) {
+    $this->src = $src;
+    $this->alt = $alt;
+    $this->name = $name;
+    $this->responsive = $responsive;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function fromArray(array $values = []): ValueObjectInterface {
+    $values += ['alt' => '', 'name' => '', 'responsive' => TRUE];
+
+    $object = new static(
+      $values['src'],
+      $values['alt'],
+      $values['name'],
+      $values['responsive']
+    );
+
+    return $object;
+  }
+
+  /**
+   * Getter.
+   *
+   * @return string
+   *   Property value.
+   */
+  public function getSource(): string {
+    return $this->src;
+  }
+
+  /**
+   * Getter.
+   *
+   * @return string
+   *   Property value.
+   */
+  public function getName(): string {
+    return $this->name;
+  }
+
+  /**
+   * Getter.
+   *
+   * @return string
+   *   Property value.
+   */
+  public function getAlt(): string {
+    return $this->alt;
+  }
+
+  /**
+   * Getter.
+   *
+   * @return bool
+   *   Property value.
+   */
+  public function isResponsive(): bool {
+    return $this->responsive;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getArray(): array {
+    return [
+      'name' => $this->getName(),
+      'src' => $this->getSource(),
+      'alt' => $this->getAlt(),
+      'responsive' => $this->isResponsive(),
+    ];
+  }
+
+  /**
+   * Construct object from a Drupal image field.
+   *
+   * @param \Drupal\image\Plugin\Field\FieldType\ImageItem $image_item
+   *   Field holding the image media.
+   * @param string $name
+   *   Name of the image media.
+   * @param bool $responsive
+   *   Whether the image should be responsive or not.
+   *
+   * @throws \Drupal\Core\TypedData\Exception\MissingDataException
+   *
+   * @return $this
+   */
+  public static function fromImageField(ImageItem $image_item, $name = '', $responsive = TRUE): ValueObjectInterface {
```

Comment:
Naming here is not correct, we are creating it from an image item, not an image field: an image field has several image items. I would also remove the responsive parameter, let's always have responsive images by default. We can remove that from the object too and always set it to true when returning the array.

We should also remove the `$name` parameters, we should be able to derive that from the `image_item`, or we leave it empty if we are not.

Reply:
I change the name and I remove the other parameters.